lua/cl_spectator_deathmatch.lua

```diff
@@ -115,75 +115,27 @@ end
-		if className == "TTTScoreboard" then
-			local old_Init = tbl.Init
-			tbl.Init = function(self)
-				old_Init(self)
-				local t = vgui.Create("TTTScoreGroup", self.ply_frame:GetCanvas())
-				t:SetGroupInfo("Spectator Deathmatch", Color(255, 127, 39, 100), GROUP_DEATHMATCH)
-				self.ply_groups[GROUP_DEATHMATCH] = t
-			end
-			local old_UpdateScoreboard = tbl.UpdateScoreboard
-			tbl.UpdateScoreboard = function(self, force)
-				if not force and not self:IsVisible() then return end
-				for k,v in pairs(player.GetAll()) do
-					if v:IsGhost() and (LocalPlayer():IsSpec() or LocalPlayer():IsActiveTraitor()) then
-						if self.ply_groups[GROUP_DEATHMATCH] and not self.ply_groups[GROUP_DEATHMATCH]:HasPlayerRow(v) then
-							self.ply_groups[GROUP_DEATHMATCH]:AddPlayerRow(v)
-						end
-					end
-				end
-				old_UpdateScoreboard(self, force)
-			end
-		elseif className == "TTTScoreGroup" then
-			tbl.AddPlayerRow = function(self, ply)
-				if (self.group == GROUP_DEATHMATCH or ScoreGroup(ply) == self.group) and not self.rows[ply] then
-					local row = vgui.Create("TTTScorePlayerRow", self)
-					row:SetPlayer(ply)
-					self.rows[ply] = row
-					self.rowcount = table.Count(self.rows)
-					-- I sincerely hate you for putting this here, spykr
-					self.rows[ply].OnMousePressed = function(s, mc) 
-						if RightClickRow and mc == MOUSE_RIGHT then 
-							RightClickRow(ply) 
-						elseif mc == MOUSE_LEFT then 
-							s:DoClick() 
-						elseif mc == MOUSE_RIGHT then
-							s:DoRightClick()
-						end
-					end
-					self:PerformLayout()
-				end
-			end
-			tbl.UpdatePlayerData = function(self)
-				local to_remove = {}
-				for k,v in pairs(self.rows) do
-					if ValidPanel(v) and IsValid(v:GetPlayer()) and ((self.group == GROUP_DEATHMATCH and v:GetPlayer():IsGhost() and (LocalPlayer():IsSpec() or LocalPlayer():IsActiveTraitor())) or ScoreGroup(v:GetPlayer()) == self.group) then
-						v:UpdatePlayerData()
-					else
-						table.insert(to_remove, k)
-					end
-				end
-				if #to_remove == 0 then return end
-				for k,ply in pairs(to_remove) do
-					local pnl = self.rows[ply]
-					if ValidPanel(pnl) then
-						pnl:Remove()
-					end
-					self.rows[ply] = nil
-				end
-				self.rowcount = table.Count(self.rows)
-				self:UpdateSortCache()
-				self:InvalidateLayout()
+	function ScoreGroupDM(p)
+		if LocalPlayer():IsTerror() and p:GetNWBool("PlayedSRound",false) and p:IsGhost() then
+			if p:GetNWBool("body_found", false) then
+				return GROUP_FOUND
+			elseif LocalPlayer():IsActiveTraitor() then
+				return GROUP_NOTFOUND
+			else
+				return GROUP_TERROR
 			end
+		elseif p:IsGhost() and (LocalPlayer():IsSpec() or LocalPlayer():IsGhost()) then
+			return GROUP_DEATHMATCH
 		end
-		return old_vguiRegister(className, tbl, base)
 	end
-	
+	hook.Add("TTTScoreGroup", "TTTScoreGroup_Ghost", ScoreGroupDM)
+	hook.Add("PostGamemodeLoaded", "PostGamemodeLoaded_Ghost", function() AddScoreGroup("DEATHMATCH") end)
```

Comment:
If AddScoreGroup gets called earlier its nil due to the function not being loaded.